pointer.go

```diff
@@ -62,12 +83,21 @@ func parse(str string) (Pointer, error) {
 // Pointer represents a parsed JSON pointer
 type Pointer []string
 
+// NewPointer creates a Pointer with a pre-allocated block of memory
+// to avoid repeated slice expansions
+func NewPointer() Pointer {
+	return make([]string, 0, defaultPointerAllocationSize)
+}
+
 // String implements the stringer interface for Pointer,
 // giving the escaped string
 func (p Pointer) String() (str string) {
 	for _, tok := range p {
 		str += "/" + escapeToken(tok)
 	}
+	if len(str) == 0 {
+		str = "/"
+	}
```

Comment:
Are we sure this matches the spec? I'm seeing two new test failures with this change. It may be that the tests are wrong.

```
=== RUN   TestParse
    TestParse: pointer_test.go:72: case 3 error mismatch. expected: 'parse ://: missing protocol scheme', got: 'parse "://": missing protocol scheme'
    TestParse: pointer_test.go:77: case 5 string output mismatch: expected: '', got: '/'
    TestParse: pointer_test.go:77: case 6 string output mismatch: expected: '', got: '/'
```

But these two used to return the empty string. The failure in test case three is also present on master, and comes from bumping up go versions